### PR TITLE
BAI-1253 Client Datacard Class

### PR DESCRIPTION
**Pull request by bw27492**
- Added new parent 'Entry' class for both models and datacards.
- Moved non-unique methods from existing model class to new parent class.
- Created new datacard class

### CODE REVIEW

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/datacard.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            client=client,
+            datacard_id=datacard_id,
+            name=res["name"],
+            description=res["description"],
+        )
+        datacard._Entry__unpack(res)
+
+        datacard.get_card_latest()
+
+        return datacard
```
Comment (GB27247) [inline lib/python/src/bailo/helper/datacard.py:91]: Can use these as protected class methods in the parent class.

Comment (GB27247) [inline lib/python/src/bailo/helper/datacard.py:91]: What's stopping me from pulling down a Model as a Datacard here. There isn't anything differentiating them apart from their kind.

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: lib/python/src/bailo/helper/datacard.py (hunk 1 above)
Comment (bw27492) [reply lib/python/src/bailo/helper/datacard.py]: Done

File: lib/python/src/bailo/helper/datacard.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        :return: A datacard object
+        """
+        res = client.get_model(model_id=datacard_id)["model"]
+        datacard = cls(
+            client=client,
+            datacard_id=datacard_id,
+            name=res["name"],
+            description=res["description"],
+        )
+        datacard._Entry__unpack(res)
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/datacard.py]: Done

File: lib/python/src/bailo/helper/datacard.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        :param data_card: Datacard dictionary, defaults to None
+
+        ..note:: If a datacard is not provided, the current datacard attribute value is used
+        """
+        self._Entry__update_card(card=data_card)
+
+    @property
+    def data_card(self):
+        return self.card
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/datacard.py]: Done

File: lib/python/src/bailo/helper/datacard.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+    @data_card.setter
+    def data_card(self, value):
+        self.card = value
+
+    @property
+    def data_card_version(self):
+        return self.card_version
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/datacard.py]: Done

File: lib/python/src/bailo/helper/datacard.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+    @data_card_version.setter
+    def data_card_version(self, value):
+        self.card_version = value
+
+    @property
+    def data_card_schema(self):
+        return self.card_schema
```
[the remark this answers is not in the document]
Comment (bw27492) [reply lib/python/src/bailo/helper/datacard.py]: Done

**Review by GB27247 — APPROVED** [review]

**Pull request merged**